lib/vsc/accounting/data/userdb.py

```diff
@@ -155,10 +162,11 @@ def user_basic_record(username):
     user_record = {'field': 'Unknown'}
 
     # Determine site of account
-    site_index = (BRUSSEL, ANTWERPEN, LEUVEN, GENT)
+    site = (None, BRUSSEL, ANTWERPEN, LEUVEN, GENT)
```

Comment:
where does this happen?

Reply:
this is the fallback code in case the VSC account page returns nothing for the user. It is used to assign a site to the user based on the VSC number

Reply:
but when does that happen? If it's a VSC user, it should return something.

Reply:
Yes, I've never seen that happen in real use cases. I sometimes use this path for testing to avoid the hassle of connecting to the VSC account page.

Reply:
You should handle that in another way, either mock and use unit testing or provide a real environment.

This is confusing and testing code should not be added in productoin code.

Reply:
The fact that I use it in testing does not mean that it is testing code. It is a fallback option in case the account page does not return the record of the user, which should never happen but it can happen.

Reply:
if the accountpage doesn't return a record, it should fail because something is very very wrong.

Reply:
I added a warning in case any VSC account is generated with `user_basic_record()`

Reply:
> if the accountpage doesn't return a record, it should fail because something is very very wrong.
I don't agree. For the accounting I do not care if the account exists or not in the account page. It is already in the ELK logs, so we have to account it in any case. Keep in mind that we only retrieve the research field and site of the user from the account page. We do not use it to validate the existence of the account on its own.

Reply:
For slurm it does matter. This cannot ever happen with slurm unless something is fundamentally wrong.

And even currently with torque: something is pretty wrong.

Reply:
Absolutely, if that happens it is an indication that something is very wrong somewhere else. But the accounting should still be generated in such a case. In the next update, I'll try to somehow flag such accounts so that they are visible in the final report.